src/account/UpgradeableModularAccount.sol

```diff
@@ -227,6 +224,11 @@ contract UpgradeableModularAccount is
         bytes4 selector = bytes4(data);
         AccountStorage storage _storage = getAccountStorage();
 
+        // Make sure plugin is allowed to spend native token.
+        if (value > 0 && value > msg.value && !_storage.pluginData[msg.sender].canSpendNativeToken) {
```

Comment:
What's the reasoning behind `value > msg.value`?
Does it mean `canSpendNativeToken` will be ignored if `value <= msg.value`?

Reply:
This check basically ensures that the plugin is only spending from the account's native token balance if it has permission to do so. If the plugin is managing its own native token balance (outside of the account), and covers the amount sent via `executeFromPluginExternal`, then it isn't deducting anything from the user's balance and therefore can send a call with value.